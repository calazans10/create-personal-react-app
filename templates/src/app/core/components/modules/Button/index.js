import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './index.scss';

const Button = ({ type, children, onClick }) => {
  const btnClasses = classNames({
    button: true,
    [`button--${type}`]: true,
  });

  return (
    <button type="button" className={btnClasses} onClick={onClick}>
      {children}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.oneOf(['default', 'primary', 'success', 'danger', 'warning']),
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

Button.defaultProps = {
  type: 'default',
};

export default Button;
