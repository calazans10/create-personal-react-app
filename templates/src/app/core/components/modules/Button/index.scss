@import 'styles/utilities';

.button {
  display: block;
  width: 100%;
  margin-bottom: 0;
  padding: rem(8) rem(4);
  border: none;
  border-radius: rem(5);
  border-color: transparent;
  background-image: none;
  color: #fff;
  font-size: em(14);
  font-weight: normal;
  line-height: 1.3333333;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  touch-action: manipulation;
  user-select: none;

  &--default {
    background-color: #bdc3c7;
  }

  &--primary {
    background-color: #00aeef;
  }

  &--success {
    background-color: #21d376;
  }

  &--danger {
    background-color: #e33b15;
  }

  &--warning {
    background-color: #f39c12;
  }
}
