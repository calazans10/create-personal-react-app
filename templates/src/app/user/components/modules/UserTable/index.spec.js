import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../../../../core/components/modules/Button';
import { UserTable } from './index';

describe('UserTable', () => {
  describe('when there are users', () => {
    it('renders without crashing', () => {
      const props = {
        users: [
          {
            id: 1,
            name: 'Leanne Graham',
            username: 'Bret',
            email: 'Sincere@april.biz',
            address: {
              street: 'Kulas Light',
              suite: 'Apt. 556',
              city: 'Gwenborough',
              zipcode: '92998-3874',
              geo: {
                lat: '-37.3159',
                lng: '81.1496',
              },
            },
            phone: '1-770-736-8031 x56442',
            website: 'hildegard.org',
            company: {
              name: 'Romaguera-Crona',
              catchPhrase: 'Multi-layered client-server neural-net',
              bs: 'harness real-time e-markets',
            },
          },
        ],
        onSelectUser: jest.fn(),
        onShowModal: jest.fn(),
      };
      const wrapper = shallow(<UserTable {...props} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('when there are no users', () => {
    it('renders nothing', () => {
      const props = {
        users: [],
        onSelectUser: jest.fn(),
        onShowModal: jest.fn(),
      };
      const wrapper = shallow(<UserTable {...props} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should handle the click event', () => {
      const props = {
        users: [
          {
            id: 1,
            name: 'Leanne Graham',
            username: 'Bret',
            email: 'Sincere@april.biz',
            address: {
              street: 'Kulas Light',
              suite: 'Apt. 556',
              city: 'Gwenborough',
              zipcode: '92998-3874',
              geo: {
                lat: '-37.3159',
                lng: '81.1496',
              },
            },
            phone: '1-770-736-8031 x56442',
            website: 'hildegard.org',
            company: {
              name: 'Romaguera-Crona',
              catchPhrase: 'Multi-layered client-server neural-net',
              bs: 'harness real-time e-markets',
            },
          },
        ],
        onSelectUser: jest.fn(),
        onShowModal: jest.fn(),
      };

      const wrapper = shallow(<UserTable {...props} />);
      wrapper
        .find(Button)
        .at(0)
        .simulate('click');
      expect(props.onSelectUser).toHaveBeenCalledTimes(1);
      expect(props.onShowModal).toHaveBeenCalledTimes(1);
    });
  });
});
