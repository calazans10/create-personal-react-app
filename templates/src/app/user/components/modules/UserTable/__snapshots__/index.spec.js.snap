// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTable when there are no users renders nothing 1`] = `
<table
  className="user-table"
>
  <thead>
    <tr>
      <th
        className="user-table__title"
      >
        ID
      </th>
      <th
        className="user-table__title"
      >
        Nome
      </th>
      <th
        className="user-table__title"
      >
        Usuário
      </th>
      <th
        className="user-table__title"
      >
        Email
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="user-table__item user-table__item--centered"
        colSpan="4"
      >
        Nenhum usuário encontrado.
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`UserTable when there are users renders without crashing 1`] = `
<table
  className="user-table"
>
  <thead>
    <tr>
      <th
        className="user-table__title"
      >
        ID
      </th>
      <th
        className="user-table__title"
      >
        Nome
      </th>
      <th
        className="user-table__title"
      >
        Usuário
      </th>
      <th
        className="user-table__title"
      >
        Email
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="1"
    >
      <td
        className="user-table__item"
      >
        1
      </td>
      <td
        className="user-table__item"
      >
        Leanne Graham
      </td>
      <td
        className="user-table__item"
      >
        Bret
      </td>
      <td
        className="user-table__item"
      >
        Sincere@april.biz
      </td>
    </tr>
  </tbody>
</table>
`;
